tasks/assign_pr.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+import re
+from collections import Counter
+
+from invoke import task
+
+from tasks.libs.pipeline_notifications import read_owners
+
+
+@task
+def add_team_labels(ctx, pr_id):
+    """
+    Add team labels to a PR based on the changed dependencies
+    """
+
+    codeowners = read_owners(".github/CODEOWNERS")
+
+    # Get what was changed according to PR title
+    title_words = ctx.run(f"gh pr view {pr_id} | head -n 1", hide=True).stdout.split()
+    dependency = title_words[2]
+    if "group" in title_words:  # dependabot defines group. Currently dep name is part of the group name
+        group_index = title_words.index("group")
+        dependency = title_words[group_index - 1]
+    import_module = re.compile(rf"^[ \t]*\"{dependency}.*$")
+    folder = f"./{title_words[-1][1:]}" if title_words[-2] == "in" else "."
+
+    # Find the responsible person for each file
+    owners = []
+    for root, _, files in os.walk(folder):
+        if root == "./.git" or any(root.startswith(prefix) for prefix in ["./venv", "./.git/"]):
+            continue
+        for file in files:
+            file_path = os.path.join(root, file)
+            if "doc" in file_path.casefold():
+                continue
+            with open(file_path, "r") as f:
+                try:
+                    for line in f:
+                        if is_go_module(dependency):
+                            if import_module.match(line):
+                                owners.extend([owner[1] for owner in codeowners.of(file_path[2:])])
```

Comment:
Why remove the first two characters of the path ? Is it `./` ?
If so you might want to use `os.path.normpath()` to normalize the path

Reply:
Indeed I can use this it's better